### ISSUE TITLE
Returning a string or an array containing strings in render functions doesn't work

### ISSUE THREAD

**Issue by nkovacs** [vuejs/vue#7681]
### Version
2.5.13

### Reproduction link
[https://jsfiddle.net/12489eae/](https://jsfiddle.net/12489eae/)

### Steps to reproduce
Write a functional component with a render function that returns a string or an array containing strings.

### What is expected?
The component renders the strings as text nodes.

### What is actually happening?
Uncaught TypeError: Cannot create property 'isRootInsert' on string 'Test'

---
This works:

```
return h("div", linesWithBreaks).children
```

I'd also like to be able to use html templates with vue-loader:

```
<template functional>
	<template v-for=`line in props.text.split("\n")`>
		{{ line }}
		<br/>
	</template>
</template>
```

But that also fails because of https://github.com/vuejs/vue-loader/issues/1168

**Comment by posva** [comment]
I think returning just a string is definitely not worth because you're better off just writing the string.
Regarding the array containing strings, we could create textnodes for them

**Comment by codebryo** [comment]
@nkovacs So the problem here is that you are not creating a root node. The output that you are returning would look like:
```html
Test
<br>
Test
<br>
Test
<br>
<br>
```

That is not valid within Vue as it requires a root element. So either wrap it into a `<p>` tag directly instead of defining that in your template. Or pass in a wrapping element tag name like so:

```html
<nl2br :text="message2" wrap="p"></nl2br>
```

and in the component: 

```js
Vue.component("nl2br", {
  functional: true,
  props:      {
    text: {
      type: String,
      required: true,
    },
    wrap: {
      type: String,
      default: 'div'
    }
  },
	render(h, context) {
		const lines = context.props.text.split("\n")
		const linesWithBreaks = []
		lines.forEach((line) => {
			linesWithBreaks.push(line, h("br"))
		})
	  return h(context.props.wrap, linesWithBreaks).children
	},
})
```

**Comment by nkovacs** [comment]
But it's a functional component, and multiple root elements are allowed there.

**Comment by codebryo** [comment]
Okay, my bad. Never used functional components in a way that I ended up with multiple roots.

**Comment by yyx990803** [comment]
This has been fixed in 2.5.14
